Gems/Maestro/Code/Source/Components/EditorSequenceAgentComponent.cpp

```diff
@@ -165,8 +173,83 @@ namespace Maestro
         AZ::EntityId curEntityId = GetEntityId();
 
         // remove this SequenceAgent from this entity if no sequenceComponents are connected to it
-        AzToolsFramework::EntityCompositionRequestBus::Broadcast(&AzToolsFramework::EntityCompositionRequests::RemoveComponents, AZ::Entity::ComponentArrayType{this});        
+        {
+            AZ_Trace("EditorSequenceAgentComponent", "DisconnectSequence(): removing self from entity %s, %s.",
+                curEntityId.ToString().c_str(), entity->GetName().c_str());
+            // The normal call for a component would be:
+            // AzToolsFramework::EntityCompositionRequestBus::Broadcast(&AzToolsFramework::EntityCompositionRequests::RemoveComponents, AZ::Entity::ComponentArrayType{this});
+            // but this one is not used, as this component is created abnormally via user actions in EditorSequenceComponent.
```

Comment:
Do you know the reason behind the fact this components are created abnormaly ? Because it sounds like that it shouldn't be if we have to 'hack' the destruction if the construction was also 'hacked'. Again I'm not too familiar with the trackview architecture in O3DE (is there some reviewers who 'owns' trackview ?)

Reply:
New Entity Ids, new Entity Aliases, new Component Ids and new Instance Ids are generated randomly, and then used as keys into unordered maps.
As an `EditorSequenceAgentComponent` is not created directly by user action, but as a result of a `EditorSequenceComponent` manipulation, - it does not need a local undo stack for its construction / destruction,  - and this local undo stack became broken when `EditorSequenceComponent` `dtor` was call before the `EditorSequenceAgentComponent` `dtor`.
One of the reasons I failed to catch this during days-and-days-and-days debugging line-after-line, is that the `EditorSequenceAgentComponent` misses explicit `ctor` and `dtor` I could put breakpoints on. Next time I'll add all missing methods and put logs into these, as I've done here (but then cleaned-up debugging code).

Does this answer you question?

Reply:
Yeah totally thanks for the explanation 👍 Twice in my career I've seen people working fulltime and struggling on a trackview feature, it does sounds like a very complex area to debug

Reply:
I'm a tad worried that this custom spawn and destroy logic will become a pain to maintain, as you'll want to have any modification done in the 'normal' logic. If the issue comes from the undo, isn't it possible to use the 'normal' codepath but asking for undo to be disabled during spawn/unspawn ?

Reply:
> isn't it possible to use the 'normal' codepath but asking for undo to be disabled during spawn/unspawn ?

Managing undo / redo state is (rightly) encapsulated, and I won't dare to mess with the inner undo / redo logic, 
I also believe that undo / redo may work during spawn/despawn when evaluating patches.

So to temporary suppress undo / redo for specific operations I've
added the new API to maintain "ignored entities list",
added checking this list when attempting to add dirty entities,
and then used this new API bracketing the the 'normal' codepath for the `EditorSequenceAgentComponent` destruction in the `EditorSequenceAgentComponent::DisconnectSequence()` (thus reverting to using the original 'normal' codepath).

Please see commit https://github.com/o3de/o3de/pull/18538/commits/191e5cff77d2c958ca09414ee2f04bc829072d66.